### DISCUSSION TITLE
"TypeError: Cannot read property '0' of undefined" after creating new single type

### DISCUSSION THREAD

**Discussion by jacobsandersen** [strapi/strapi#7224] [Legacy Topics]
Hi there, I just installed Strapi to check it out. I made a single type for a conceptual "resume".

![like this](https://i.imgur.com/DxulgEf.png)

However, when going to edit it to fill out the content I get this is in my logs:
```
[2020-07-28T08:01:06.436Z] error TypeError: Cannot read property '0' of undefined
    at markArraySubdocsPopulated (/srv/app/node_modules/mongoose/lib/document.js:559:48)
    at model.Document.$__init (/srv/app/node_modules/mongoose/lib/document.js:519:3)
    at model.syncWrapper [as $__init] (/srv/app/node_modules/kareem/index.js:234:23)
    at model.Document.init (/srv/app/node_modules/mongoose/lib/document.js:485:8)
    at completeMany (/srv/app/node_modules/mongoose/lib/helpers/query/completeMany.js:41:14)
    at /srv/app/node_modules/mongoose/lib/query.js:1952:9
    at /srv/app/node_modules/mongoose/lib/model.js:4759:16
    at /srv/app/node_modules/mongoose/lib/utils.js:276:16
    at /srv/app/node_modules/mongoose/lib/model.js:4780:21
    at next (/srv/app/node_modules/mongoose/lib/model.js:4230:5)
    at _done (/srv/app/node_modules/mongoose/lib/model.js:4362:5)
    at _next (/srv/app/node_modules/mongoose/lib/model.js:4352:7)
    at /srv/app/node_modules/mongoose/lib/model.js:4759:16
    at /srv/app/node_modules/mongoose/lib/utils.js:276:16
    at /srv/app/node_modules/mongoose/lib/model.js:4780:21
    at /srv/app/node_modules/mongoose/lib/query.js:4379:11
```

Then, the page just loads forever.

How can I find out more about what's going on?

**Comment by jacobsandersen** [comment]
Seems to be related to the repeatable component. If I make it a single component, then it works.

**Comment by lauriejim** [comment]
Hello! Can you peals share your application, it will be easier to reproduce your data structure.
Thank you.

**Comment by jacobsandersen** [comment]
@lauriejim I've just solved it.

I first created this `Resume` type without any custom components. I was using Text, Rich Text, and Email only. Then, I added some content.

After having done that, I removed the Rich Text components for ProfessionalExperience and Education and then using the same names created them as components.

At that point, this error occurred.

I can only assume that the content was not removed from the database when the actual structure (Resume) was modified. This would mean that when the ProfessionalExperience/Education was re-added to the Resume, the editor was expecting either nothing or data that matched the components.

There was simply a string in there, so it crashed.

Maybe.

In any case, I solved it by reinstalling Strapi entirely (such that I could recreate the structure with no content behind it) so I cannot send you the actual structure. I could try to reproduce it if you wish.

**Comment by lauriejim** [comment]
Ho okay, so in fact it's related to this issue https://github.com/strapi/strapi/issues/1114

**Comment by jacobsandersen** [comment]
@lauriejim Is there a workaround for this short of reinstalling? If so, I'm ok waiting for a fix otherwise it seems kind of major?

Like, manually clearing out the data from the db? I do like strapi, it's very clever.
